huaweicloud/services/acceptance/css/resource_huaweicloud_css_logstash_custom_certificate_test.go

```
@@ -0,0 +1,144 @@
+package css
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getLogstashCertificateFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	cssV1Client, err := cfg.CssV1Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating CSS v1 client: %s", err)
+	}
+
+	getCertificateHttpUrl := "v1.0/{project_id}/clusters/{cluster_id}/certs/{cert_id}"
+	getCertificatePath := cssV1Client.Endpoint + getCertificateHttpUrl
+	getCertificatePath = strings.ReplaceAll(getCertificatePath, "{project_id}", cssV1Client.ProjectID)
+	getCertificatePath = strings.ReplaceAll(getCertificatePath, "{cluster_id}", state.Primary.Attributes["cluster_id"])
+	getCertificatePath = strings.ReplaceAll(getCertificatePath, "{cert_id}", state.Primary.ID)
+
+	getCertificatePathOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+
+	getCertificateResp, err := cssV1Client.Request("GET", getCertificatePath, &getCertificatePathOpt)
+	if err != nil {
+		return nil, golangsdk.ErrDefault404{}
```

it is better to return the error directly